electron/launcher/main-process/helpers/login.js

```diff
@@ -0,0 +1,203 @@
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const path = require('path');
+const keytar = require('keytar');
+const app = require('electron').app;
+const aes256 = require('helpers/aes256');
+
+const config = require('config.json');
+const secret = config.SECRET;
+const djangoUrl = config.DJANGO_SERVER.HOST +":" +config.DJANGO_SERVER.PORT;
+
+// REGULAR LOGIN WITH USERNAME AND PASSWORD
+const login = cred => {
+  return new Promise((resolve, reject) => {
+    let options = {
+      method: 'POST',
+      uri: djangoUrl +'/accounts/login/',
+      header: {'content-type': 'application/x-www-form-urlencoded'},
+      form: cred,
+      resolveWithFullResponse: true,
+      json: true
+    }
+    
+    let body;
+    let encryptedPassword;
+
+    aes256.encrypt(cred.password).then(encrypted => {
+      encryptedPassword = encrypted;
+    })
+    // SEND POST REQUEST TO LOGIN
+    .then(() => {
+      return request(options).then(response => {
+        body = response.body;
+        return response.headers['set-cookie'];
+      })
+      .catch(error => {
+        throw error.error;
+      });
+    })
+    // PARSE Set-Cookie HEADER INTO COOKIE OBJECTS
+    .then(cookieStrings => {
+      let promises = [];
+
+      for(cookieString of cookieStrings){
+        promises.push(new Promise((resolve, reject) => {
+          resolve(cookie.parse(cookieString));
+        }));
+      }
+
+      return Promise.all(promises).then(cookies => {
+        return cookies;
+      });
+    })
+    // SAVES USERNAME, PASSWORD, CSRF TOKEN, SESSION TOKEN IN OPERATING SYSTEM'S PASSWORD VAULT
+    .then(cookies => {
+      let promises = [];
+      let expiry = 0;
+
+      promises.push(keytar.setPassword('Dietstory', body.name, encryptedPassword));
+      for(cookieObject of cookies){
+        if(cookieObject.hasOwnProperty('csrftoken')){
+          csrfToken = cookieObject.csrftoken;   // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_CSRF', body.name, cookieObject.csrftoken));
+        }
+        else if(cookieObject.hasOwnProperty('sessionid')){
+          sessionKey = cookieObject.sessionid;  // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_Session', body.name, cookieObject.sessionid));
+          expiry = Date.parse(cookieObject.expires);
+        }
+      }
+      
+      return Promise.all(promises).then(() => {
+        return expiry;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // SAVE USER INFO FOR FUTURE AUTO-LOGIN TO USE (ENCRYPTED)
+    .then(expiry => {
+      let encrypt = string => {
+        return new Promise((resolve, reject) => {
+          let cipher = crypto.createCipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let encrypted = cipher.update(string, 'utf8', 'hex');
+          encrypted += cipher.final('hex');
+          resolve(encrypted);
+        });
+      }
```

Comment:
is this `encrypt` definition just `aes256.encrypt(string)`? Perhaps use that method instead of redefining it

Reply:
I moved those encrypt/decrypt functions out into a different file because I wanted to reuse them elsewhere. Looks like I forgot to remove them from their original location and change the code that uses them.